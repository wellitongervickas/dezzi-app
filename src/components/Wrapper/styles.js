import styled from 'styled-components';

import { sm } from 'assets/stylesheets/js/sizes';
import { white } from 'assets/stylesheets/js/colors';

export const WrapperContainer = styled.div`
  background-color: ${(props) => props.backgroundColor || white};
  padding: ${sm()};
  border-radius: ${sm()};
`;

export default WrapperContainer;
