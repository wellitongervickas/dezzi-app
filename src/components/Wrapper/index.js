import React from 'react';
import PropTypes from 'prop-types';

import { white } from 'assets/stylesheets/js/colors';

import { WrapperContainer } from 'components/Wrapper/styles';

const Wrapper = ({ backgroundColor, children }) => (
  <WrapperContainer
    backgroundColor={backgroundColor}
  >
    {children}
  </WrapperContainer>
);

Wrapper.defaultProps = {
  backgroundColor: white,
  children: null,
};

Wrapper.propTypes = {
  children: PropTypes.node,
  backgroundColor: PropTypes.string,
};

export default Wrapper;
