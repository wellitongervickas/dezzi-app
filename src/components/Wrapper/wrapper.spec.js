import React from 'react'
import { render, cleanup } from '@testing-library/react';
import Wrapper from '.';

describe('Components Wrapper', () => {
  afterEach(cleanup);

  it('should have a snapshot with a default wrapper component', async () => {
    const { asFragment } = render(<Wrapper />);
    expect(asFragment(<Wrapper />)).toMatchSnapshot();
  });
});
