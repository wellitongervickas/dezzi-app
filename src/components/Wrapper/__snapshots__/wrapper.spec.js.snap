// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Wrapper should have a snapshot with a default wrapper component 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm gpZwYE"
  />
</DocumentFragment>
`;
